### PR TITLE
Add err popups

### PR DESCRIPTION
**Pull request by LadyMopar**

### CODE REVIEW

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 1]
```diff
@@ -357,10 +357,10 @@ private void handleDeleteTransactionFromTableView() {
                 task.startTask();
                 task.waitForComplete();
             } else {
-                System.out.println("No transactions deleted. Selected index: " + selectedIndex);
+                setupErrorPopup("No transactions deleted. Selected index.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Why still have the " Selected index."

File: src/main/java/ledger/user_interface/ui_controllers/TransactionPopupController.java [hunk 2]
```diff
@@ -125,8 +125,7 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
 
                 task.startTask();
             } catch (Exception e) {
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/TransactionPopupController.java]: Not part of code you changed but that should not be Catch Exception

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 3]
```diff
@@ -357,10 +357,10 @@ private void handleDeleteTransactionFromTableView() {
...
             }
         } catch (StorageException e) {
-            System.out.println("Error deleting transaction: " + e.getMessage());
+            setupErrorPopup("Error deleting transaction.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should  probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 4]
```diff
@@ -342,7 +342,7 @@ private void updateTransactionTableView() {
             this.transactionTableView.setItems(observableTransactionModels);
 
         } catch (StorageException e) {
-            System.out.println("Error of loading all transactions into list view: " + e.getMessage());
+            setupErrorPopup("Error loading all transactions into list view.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 5]
```diff
@@ -244,7 +244,7 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
 
                 updateTransactionTableView();
             } catch (StorageException e) {
-                System.out.println("Error editing transaction amount: " + e.getMessage());
+                setupErrorPopup("Error editing transaction amount.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 6]
```diff
@@ -232,7 +232,7 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
                 } else if (pendingToSetString.equals("Pending")) {
                     pendingToSet = true;
                 } else {
-                    System.out.print("Transaction pending status not updated. Invalid input.");
+                    setupErrorPopup("Transaction pending status not updated. Invalid input.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Say what are the valid inputs (Cleared/Pending)

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 7]
```diff
@@ -214,7 +214,7 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
 
                 updateTransactionTableView();
             } catch (StorageException e) {
-                System.out.println("Error editing transaction payee: " + e.getMessage());
+                setupErrorPopup("Error editing transaction payee");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 8]
```diff
@@ -175,7 +175,7 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
 
                 updateTransactionTableView();
             } catch (StorageException e) {
-                System.out.println("Error editing transaction payee: " + e.getMessage());
+                setupErrorPopup("Error editing transaction payee.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 9]
```diff
@@ -143,7 +143,7 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
...
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 10]
```diff
@@ -109,9 +109,9 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
 
                 updateTransactionTableView();
             } catch (StorageException e) {
-                System.out.println("Error editing transaction date: " + e.getMessage());
+                setupErrorPopup("Error editing transaction date.");
             } catch (ParseException e) {
-                System.out.println("Error parsing date string: " + e.getMessage());
+                setupErrorPopup("Error parsing date string.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 11]
```diff
@@ -109,9 +109,9 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
 
                 updateTransactionTableView();
             } catch (StorageException e) {
-                System.out.println("Error editing transaction date: " + e.getMessage());
+                setupErrorPopup("Error editing transaction date.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/MainPageController.java [hunk 12]
```diff
@@ -85,7 +85,7 @@ public void handle(CellEditEvent<TransactionModel, String> t) {
 
                 updateTransactionTableView();
             } catch (StorageException e) {
-                System.out.println("Error editing transaction amount: " + e.getMessage());
+                setupErrorPopup("Error editing transaction amount.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/MainPageController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/AccountPopupController.java [hunk 13]
```diff
@@ -72,6 +73,7 @@ private void insertDone() {
     }
 
     private void insertFail(Exception e) {
+        this.setupErrorPopup("Account insertion error.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/AccountPopupController.java]: Should probably keep the message. Latter definitely want to translate it but even the bare message should be useful.

File: src/main/java/ledger/user_interface/ui_controllers/AccountPopupController.java [hunk 14]
```diff
@@ -61,6 +61,7 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
                 task.startTask();
             } catch (Exception e) {
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/AccountPopupController.java]: This should not be Catch Exception

File: src/main/java/ledger/user_interface/ui_controllers/AccountPopupController.java [hunk 15]
```diff
@@ -61,6 +61,7 @@ public void initialize(URL fxmlFileLocation, ResourceBundle resources) {
...
                 System.out.println("Error on account submission: " + e);
+                this.setupErrorPopup("Error on account submission - please check that all fields have valid values.");
```
Comment (cjm721) [inline src/main/java/ledger/user_interface/ui_controllers/AccountPopupController.java]: This catch does not catch error on submission but rather error on getting account or balance. Both of those methods should catch their own errors also

**Comment by taylerhow** [comment]
@shellajt What part of the code are you referring to?

**Comment by shellajt** [comment]
@taylerhow Everywhere we create an error box it looks something like "Helpful error message" + e

**Comment by cjm721** [comment]
Would like someone else to review this also as I added some code to this.

**Pull request merged**